manifests/producer/service.pp

```diff
@@ -37,16 +37,7 @@
     }
     $environment = deep_merge($env_defaults, $env)
 
-    if $facts['service_provider'] == 'systemd' {
-      fail('Console Producer is not supported on systemd, because the stdin of the process cannot be redirected')
-    } else {
-      file { "/etc/init.d/${service_name}":
-        ensure  => file,
-        mode    => '0755',
-        content => template('kafka/init.erb'),
-        before  => Service[$service_name],
-      }
-    }
+    fail('Console Producer is not supported on systemd, because the stdin of the process cannot be redirected')
```

Comment:
I am wondering if this is still valid

Reply:
Thats what I have been wondering, too. But I am not too familiar with kafka to make a call there.

Reply:
I'd vote for keeping it as `fail()`.. then I'd vote to drop both `kafka::consumer` and `kafka::producer` classes as those are mostly useless from my point of view. Though, I might be wrong 🤷🏻‍♂️

Reply:
I see 2 Options here:
1. Kafka producers are now supported for systemd by kafka -> this has to be patched to follow this
2. Kafka producers still dont support systemd -> the producer class class can be dropped
Both requires knowledge about kafka is dont have yet. Might be something we should solve in a different PR (maybe someone using kafka stumbles upon this module and helps out). For the time being we can keep the fail() part as we change nothing about the original logic (and therefore dont break stuff that hasnt been broken).

As for the consumer: I want to provide a PR that at least combines the subclasses together. The class in and of itself is usefull. Supporting Kafka consumers is a good feature.

Reply:
As far as I see there is nothing to do from the Kafka side. All the "magic" was implemented in the init-script completely (see `kafka::producer::input` IIRC). Theoretically, it's possible to implement something alike with the `StandartInput{,Text,Data}` unit params.

Fun enough, there is no "opposite way" for the `kafka::consumer`. That's why I cannot understand what is the use-case for the consumer class 🤔